cli/analytics.go

```diff
@@ -27,18 +25,12 @@ func redactStepInputs(environment map[string]string, inputs []models.Environment
 			continue
 		}
 
-		src := bytes.NewReader([]byte(inputValue))
-		dstBuf := new(bytes.Buffer)
-		secretFilterDst := filterwriter.New(secrets, dstBuf)
-
-		if _, err := io.Copy(secretFilterDst, src); err != nil {
-			return map[string]string{}, fmt.Errorf("failed to redact secrets, stream copy failed: %s", err)
-		}
-		if _, err := secretFilterDst.Flush(); err != nil {
-			return map[string]string{}, fmt.Errorf("failed to redact secrets, stream flush failed: %s", err)
+		redacted, err := analytics.RedactStringWithSecret(inputValue, secrets)
```

Comment:
This code has been extracted to analytics package as it is resued in the input redactor